src/modals/modal-overlay/modal-overlay.component.js

```diff
@@ -0,0 +1,136 @@
+              return ctrl.hideCloseIcon;
+            },
+            onBackgroundClick: function() {
+              return ctrl.onBackgroundClick;
+            },
+            isForm: function() {
+              return ctrl.isForm;
+            },
+            modalBodyTemplate: function() {
+              return ctrl.modalBodyTemplate;
+            },
+            modalBodyScope: function() {
+              return ctrl.modalBodyScope;
+            },
+            actionButtons: function () {
+              return ctrl.actionButtons;
+            }
+          }
+        })
+          .result.then(
+          function (dismissCause) {
+            ctrl.close({'dismissCause': dismissCause}); // closed
+          },
+          function (dismissCause) {
+            ctrl.close({'dismissCause': dismissCause}); // dismissed
+          }
+        );
+      };
+
+      ctrl.$onInit = function () {
+        if (ctrl.showModal === undefined) {
+          ctrl.showModal = false;
+        }
+      };
+
+      ctrl.$onChanges = function (changesObj) {
+        if (changesObj.showModal && changesObj.showModal.currentValue === true) {
+          ctrl.open();
+        }
+      };
+    }
+  });
+
+
+angular.module('patternfly.modals').component('pfModalOverlayContent', {
+  templateUrl: 'modal-overlay-template.html',
+  bindings: {
+    resolve: '<',
+    close: '&',
+    dismiss: '&'
+  },
+  controller: function ($scope) {
+    'use strict';
+    var ctrl = this;
+
+    ctrl.$onInit = function () {
+      ctrl.modalId = ctrl.resolve.modalId;
+      ctrl.titleId = ctrl.resolve.titleId || "modalTitle";
+      ctrl.modalTitle = ctrl.resolve.modalTitle;
+      ctrl.hideCloseIcon = ctrl.resolve.hideCloseIcon || false;
+      ctrl.modalBodyTemplate = ctrl.resolve.modalBodyTemplate;
+      ctrl.modalBodyScope = ctrl.resolve.modalBodyScope;
+      ctrl.actionButtons = ctrl.resolve.actionButtons;
+      ctrl.isForm = ctrl.resolve.isForm;
+      ctrl.onBackgroundClick = ctrl.resolve.onBackgroundClick;
+
+      ctrl.ok = function (label, actionFn) {
+        if (typeof actionFn === "function") {
+          // Ok button function must return 'true' or 'false'
+          // indicating whether the modal can be closed.
+          var canClose = actionFn();
+          if (canClose) {
+            ctrl.close({$value: label});
+          }
+        } else {
+          ctrl.close({$value: label});
+        }
+      };
+
+      ctrl.cancel = function (actionFn) {
+        if (typeof actionFn === "function") {
+          actionFn();
+        }
+        ctrl.dismiss({$value: 'cancel'});
+      };
+
+      $scope.$on('modal.closing', function(event, reason, closed) {
+        if (reason === "backdrop click" || reason === "escape key press") {
+          event.preventDefault();
+          if ( typeof ctrl.onBackgroundClick === "function" && ctrl.onBackgroundClick()) {
+            ctrl.dismiss({$value: 'backdropClick'});
+          } else {
+            $scope.$digest();
```

Comment:
Pls add comment `onBackgroundClick' returned false, may have changed ctrl vars; execute $digest to refresh view`